### DISCUSSION TITLE
Como fazer um login com GraphQL?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1893] [Perguntas e Respostas]
Fala galera, tudo bom?

Estou começando a me aprofundar em graphql, porém ainda tem algumas coisas nebulosas, como o login, passar e ler os headers e etc...

Sei que poderia jogar no google a pergunta acima e ver um monte de tutoriais de pessoas que nunca colocaram graphql em produção, assim como acontece na maioria dos tutoriais/artigos que encontramos sobre uma boa parte de techs...

Quero opinião de alguém que esteja usando em projetos mid-tiers reais, como fazem a parte de login? Ou usam REST para login e demais endpoints e GraphQL para a parte de buscas/catalogo?

Muito obrigado, 
abraços!

**Comment by victorferreira** [comment]
Assim como você teria _endpoints_ para autenticação, pode ter _mutations_ que cuidam da lógica de login/logout da sua aplicação. O que eu tenho feito é ter uma _mutation_ que retorna um JWT que é adicionado no header de todas as requisições subsequentes. Daí, dependendo de como o seu backend funciona, você teria alguma espécie de _middleware_ que valide o _token_ da mesma forma que faria com REST.
